### DISCUSSION TITLE
Concrete implementation of intraprocedural Analysis

### DISCUSSION THREAD

**Discussion by lbettels** [soot-oss/SootUp#736] [Q&A] [answered]
I am quite unsure what a subclass of the ForwardFlowAnalysis expects for the initialFlow() and especially the flowThrough() method as far as the concrete implementation goes.
I would be grateful if someone could hint me to what types the methods expect or even better a small implementation example i.e. for live variable analysis as this should be the easiest.

**Comment by virusdefender** [comment, marked as answer]
you can read the related part of soot document https://github.com/soot-oss/soot/wiki/Implementing-an-intra-procedural-data-flow-analysis-in-Soot

**Reply by lbettels** [reply]
Thanks for the answer! I was able to figure it out by taking the [ArraySparseSet.java](https://github.com/soot-oss/soot/blob/develop/src/main/java/soot/toolkits/scalar/ArraySparseSet.java), the [FlowSet.java](https://github.com/soot-oss/soot/blob/develop/src/main/java/soot/toolkits/scalar/FlowSet.java) and the [AbstractFlowSet.java](https://github.com/soot-oss/soot/blob/develop/src/main/java/soot/toolkits/scalar/AbstractFlowSet.java) from the original Soot project. I plan to share my sample code in the next weeks over at the [Show and Tell](https://github.com/soot-oss/SootUp/discussions/categories/show-and-tell) discussion board as reference for other people.
Cheers!
